### ISSUE TITLE
Short title text for \tcblistof

### ISSUE THREAD

**Issue by dbitouze** [T-F-S/tcolorbox#124]
`\tcblistof[⟨macro⟩]{⟨name⟩}{⟨title text⟩}` displays the generated list of `tcolorbox`es with the given ⟨name⟩. The heading is generated by `⟨macro⟩{⟨title text⟩}` where `\section` is the default setting for ⟨macro⟩. So far, so good. 

But it could be nice to provide instead:
```latex
\tcblistof[⟨macro⟩]{⟨name⟩}[⟨short title text⟩]{⟨title text⟩}
```
in order to let the heading be generated by `⟨macro⟩[⟨short title text⟩]{⟨title text⟩}`, as in `\section[⟨short title text⟩]{⟨title text⟩}`.

**Comment by muzimuzhi** [comment]
From the definition of `\tcblistof`, the only requirement for `<macro>` is that it accepts one mandatory argument `<title text>`, so you can just use
```tex
\tcblistof[{\section[short title]}]{tcb}{title text}
```

https://github.com/T-F-S/tcolorbox/blob/f5a6540c0f35743853dea8e761d073ce080595ac/tex/latex/tcolorbox/tcolorbox.sty#L2108-L2114

A full example
```tex
\documentclass{article}
\usepackage{tcolorbox}
\newtcolorbox[list inside=tcb]{mytcb}[1][]{#1}

\pagestyle{headings}

\begin{document}
\tcblistof[{\section[short title]}]{tcb}{title text}

\begin{mytcb}[title=One]
  content
\end{mytcb}

\begin{mytcb}[title=Two]
  content
\end{mytcb}
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/95889575-23745400-0db5-11eb-9904-724611119555.png)

**Comment by dbitouze** [comment]
Indeed, I was pretty sure such a workaround was possible :smile: But I didn't test it since:

- it may be puzzling,
- not aware users are likely to forget the curly braces around `\section[short title]` and may be puzzled :wink:,
- an optional argument as I suggested seems as natural as `\section[⟨short title text⟩]{⟨title text⟩}`.

**Comment by muzimuzhi** [comment]
`\tcblistof[{\section[short title]}]{tcb}{title text}` is tricky and more or less a short-term workaround.

BTW, if `[⟨short title text⟩]{⟨title text⟩}` is added, then perhaps `*[⟨title text⟩]` should be supported too? In `article` class, `\listof(figures|tables)` both use `\section*`, followed by a manual `\markboth`.

**Comment by dbitouze** [comment]
For the starred version, I agree. Or, at least, the documentation should stipulate that, in:
```latex
\tcblistof[⟨macro⟩]{⟨name⟩}[⟨short title text⟩]{⟨title text⟩}
```
starred versions of ⟨macro⟩ are allowed.

**Comment by T-F-S** [comment]
Currently, I have not much time, but for a latter version I can add at least some examples to the documentation.

**Comment by T-F-S** [comment]
I just implemented an enhanced version of `\tcblistof` for the coming next version of `tcolorbox`. It will support an optional short title.

**Comment by T-F-S** [comment]
Implemented with version 4.50 (2021/05/21)
